services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/application/ApplicationServiceTest.kt

```diff
@@ -119,383 +155,392 @@ class ApplicationServiceTest {
-    ) {
-        val hankeEntity =
-            HankeEntity(hankeTunnus = HANKE_TUNNUS, id = 1, nimi = HankeFactory.defaultNimi)
-        val applicationEntity =
-            AlluDataFactory.createApplicationEntity(
-                id = 3,
-                alluid = null,
-                userId = USERNAME,
-                applicationData = applicationData.copy(rockExcavation = rockExcavation),
-                hanke = hankeEntity,
-            )
-        val sender = HankeKayttajaFactory.createEntity()
-        every { applicationRepo.findOneById(3) } returns applicationEntity
-        every { applicationRepo.save(any()) } answers { firstArg() }
-        every { geometriatDao.calculateCombinedArea(any()) } returns 100f
-        every { geometriatDao.calculateArea(any()) } returns 100f
-        every { geometriatDao.isInsideHankeAlueet(1, any()) } returns true
-        every { cableReportService.create(any()) } returns 852
-        justRun { cableReportService.addAttachment(852, any()) }
-        every { cableReportService.getApplicationInformation(852) } returns
-            AlluDataFactory.createAlluApplicationResponse(852)
-        justRun { attachmentService.sendInitialAttachments(852, any()) }
-        every { hankeKayttajaService.getKayttajaByUserId(1, USERNAME) } returns sender
-        justRun { emailSenderService.sendApplicationNotificationEmail(any()) }
-
-        applicationService.sendApplication(3, USERNAME)
-
-        val expectedApplicationData =
-            applicationData.copy(pendingOnClient = false, rockExcavation = rockExcavation)
-        val expectedAlluData =
-            expectedApplicationData
-                .toAlluData(HANKE_TUNNUS)
-                .copy(workDescription = applicationData.workDescription + "\n" + expectedSuffix)
-        verifySequence {
-            applicationRepo.findOneById(3)
-            geometriatDao.isInsideHankeAlueet(1, any())
-            geometriatDao.calculateCombinedArea(any())
-            geometriatDao.calculateArea(any())
-            cableReportService.create(expectedAlluData)
-            disclosureLogService.saveDisclosureLogsForAllu(expectedApplicationData, Status.SUCCESS)
-            cableReportService.addAttachment(852, any())
-            cableReportService.getApplicationInformation(852)
-            hankeKayttajaService.getKayttajaByUserId(1, USERNAME)
-            hankeKayttajaService.saveNewTokensFromApplication(
-                any(),
-                hankeEntity.id!!,
-                hankeEntity.hankeTunnus!!,
-                hankeEntity.nimi!!,
-                USERNAME,
-                sender
-            )
-            emailSenderService.sendApplicationNotificationEmail(any())
-            emailSenderService.sendApplicationNotificationEmail(any())
-            applicationRepo.save(any())
+        @ParameterizedTest
+        @CsvSource("true,Louhitaan", "false,Ei louhita")
+        fun `when sending should add rock excavation information to work description`(
```

Comment:
Does every test name have to start with 'when'? Even if there's no condition specified?

Reply:
The pattern I'm trying to achieve:

```
GIVEN a context -> this can usually be omitted if we have nested classes (it is the context)
WHEN some condition 
THEN expect some output
```

Test naming should be consistent. This may not be perfect but it is better than 95 % of haitaton test names. Maybe we should decide what is the way to name tests in haitaton.

Reply:
Yes, we should discuss how tests should be named.